src/frontend/src/components/Posts/Posts.jsx

```diff
@@ -1,37 +1,217 @@
+    setNumPages(numPages + 1);
+  }
+
+  const getPostsCount = useCallback(async () => {
+    if (!shouldCheckForNewPosts) {
+      return null;
+    }
+    try {
+      const res = await fetch(`${telescopeUrl}/posts`, { method: 'HEAD' });
+      if (!res.ok) {
+        throw new Error(res.statusText);
+      }
+      return res.headers.get('x-total-count');
+    } catch (error) {
+      console.log(error);
+    } finally {
+      setShouldCheckForNewPosts(false);
+    }
+    return null;
+  }, [telescopeUrl, shouldCheckForNewPosts]);
+
+  const callback = useCallback(async () => {
+    setCurrentNumPosts(await getPostsCount());
+  }, [getPostsCount]);
+
+  useEffect(() => {
+    savedCallback.current = callback;
+  });
+
+  useEffect(() => {
+    async function setPostsInfo() {
+      try {
+        setInitNumPosts(await getPostsCount());
+      } catch (error) {
+        console.log({ error });
+      }
+    }
+    setPostsInfo();
+  }, [getPostsCount, currentNumPosts]);
+
+  useEffect(() => {
+    function getCurrentNumPosts() {
+      setShouldCheckForNewPosts(true);
+      savedCallback.current();
+    }
+    savedCallback.current = callback;
+    // Polls every 5 minutes
+    const interval = setInterval(() => {
+      getCurrentNumPosts();
+    }, 5 * 60 * 1000);
+    return () => clearInterval(interval);
+  }, [callback]);
+
+  useEffect(() => {
+    function getCurrentNumPosts() {
+      setShouldCheckForNewPosts(true);
+      savedCallback.current();
+    }
+    savedCallback.current = callback;
+    // Polls every 5 minutes
+    const interval = setInterval(getCurrentNumPosts, 5 * 60 * 1000);
+    return () => clearInterval(interval);
+  }, [callback]);
+
+  useEffect(() => {
+    // Prevents alert from appearing upon page loading.
+    // Also checks whether there are new posts available
+    if (!loading && currentNumPosts !== initNumPosts && currentNumPosts !== 0) {
+      setAlert(true);
+    } else {
+      setAlert(false);
+    }
+  }, [currentNumPosts, initNumPosts, loading]);
+
+  function GenerateLoadButtonContent() {
+    if (endOfPosts) {
+      return 'No more posts.  Your turn! Add your feed...';
+    }
+    return 'Load More Posts';
+  }
 
   return posts.length > 0 ? (
-    <Container className={classes.root}>
-      {posts.map(({ id, feed, html, title, url, updated }) => (
-        <Post
-          key={id}
-          id={id}
-          author={feed.author}
-          url={url}
-          html={html}
-          title={title}
-          date={updated}
-          link={feed.link}
-        />
-      ))}
+    <Container>
+      <Container className={classes.root}>
+        {posts.map(({ id, feed, html, title, url, updated }) => (
+          <Post
+            key={id}
+            id={id}
+            author={feed.author}
+            url={url}
+            html={html}
+            title={title}
+            date={updated}
+            link={feed.link}
+          />
+        ))}
+      </Container>
+      <Container>
```

Comment:
In another PR, we should probably refactor this into its own component.

Reply:
Agreed. Too much going on for the component itself. This is still a good iteration to `index.js`